### PR TITLE
Fix thumbnail generation for command line

### PR DESCRIPTION
**Pull request by mschop**
I was asked by @mitelg to recreate this PR: https://github.com/shopware/shopware/pull/3748
Lookup that PR for details.

### CODE REVIEW

**Comment by mitelg** [comment]
hey @mschop 
you need to sign the CLA, before we could go on :slightly_smiling_face:

File: src/Core/Content/Media/Message/GenerateThumbnailsHandler.php [hunk 1]
```diff
@@ -29,10 +30,11 @@ public function __construct(
     public function __invoke(GenerateThumbnailsMessage|UpdateThumbnailsMessage $msg): void
     {
         $context = $msg->getContext();
+        $context->addState(MediaDeletionSubscriber::SYNCHRONE_FILE_DELETE);
 
         $criteria = new Criteria();
         $criteria->addAssociation('mediaFolder.configuration.mediaThumbnailSizes');
-        $criteria->addFilter(new EqualsAnyFilter('media.id', $msg->getMediaIds()));
+	    $criteria->addFilter(new EqualsAnyFilter('media.id', $msg->getMediaIds()));
```
Comment (marcelbrode) [inline src/Core/Content/Media/Message/GenerateThumbnailsHandler.php:37]: You did it again :smile:

**Comment by phizab** [comment]
I have just stumbled across the same problem and I solved it by adding the state `MediaDeletionSubscriber::SYNCHRONE_FILE_DELETE` in `ThumbnailService::updateThumbnails()` on the following line: 
https://github.com/shopware/shopware/blob/024c1f126cb7bc703aa436a18fae2c859fc9ae58/src/Core/Content/Media/Thumbnail/ThumbnailService.php#L186
I think the real problem lies in this place, because deleting the thumbnails and then directly generating them afterwards should always be synchronous, even if the method `updateThumbnails` is called from an other place than `GenerateThumbnailsCommand` or  `GenerateThumbnailsHandler`. In method `ThumbnailService::generate()` it's solved similar. What do you think?

Code:
```
$update = $this->connection->transactional(function () use ($delete, $media, $config, $context, $toBeCreatedSizes): array {
    return $context->state(function () use ($delete, $media, $config, $context, $toBeCreatedSizes): array {
        $this->thumbnailRepository->delete($delete, $context);

        $updated = $this->generateAndSave($media, $config, $context, $toBeCreatedSizes);

        $this->indexer->handle(new MediaIndexingMessage([$media->getId()]));

        return $updated;
    }, EntityIndexerRegistry::DISABLE_INDEXING, MediaDeletionSubscriber::SYNCHRONE_FILE_DELETE);
});
```

File: src/Core/Content/Media/Message/GenerateThumbnailsHandler.php (hunk 1 above)
Comment (marcelbrode) [reply src/Core/Content/Media/Message/GenerateThumbnailsHandler.php:37]: Still not resolved btw

**Comment by marcelbrode** [comment]
Hi there! Any updates on this ticket?

**Comment by marcelbrode** [comment]
Since we noticed no changes in the last 4 weeks, I'll close this ticket. Feel free to ping me, if you've rebased your branch and provided the changes asked and I will gladly reopen this ticket. Thank you :smile:

**Pull request closed without merging**